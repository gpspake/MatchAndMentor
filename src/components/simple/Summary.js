import React from 'react';
import { View, Text, Platform, TextInput, Keyboard } from 'react-native';
import Button from './Button';
import {
  updateMentorSummary,
  submitMentorSummary,
  updateMenteeSummary,
  submitMenteeSummary,
} from '../../actions';

const styles = {
  container: {
    flex: 1,
    backgroundColor: '#fff',
    justifyContent: 'space-between',
    padding: 20,
  },
  buttonStyle: {
    backgroundColor: '#55A838',
    margin: 20,
    height: 45,
    borderRadius: 10,
    justifyContent: 'center',
  },
  textStyle: {
    color: 'white',
    alignSelf: 'center',
    fontSize: 24,
    fontFamily: 'Montserrat-Regular',
  },
  headerText: {
    color: 'black',
    fontSize: 24,
    fontFamily: 'Montserrat-Light',
  },
  inputStyle: {
    borderBottomColor: 'black',
    borderBottomWidth: Platform.OS === 'ios' ? 1 : 0,
    margin: 20,
    fontSize: 24,
    fontFamily: 'Montserrat-Light',
    height: '50%',
    backgroundColor: '#e9e9e9',
    padding: 10,
  },
};

function submit(dispatch, isMentor, value, loadMenteeScreenNext, menteeOnly) {
  Keyboard.dismiss();
  if (isMentor) {
    dispatch(submitMentorSummary(value, loadMenteeScreenNext));
  } else {
    dispatch(submitMenteeSummary(value, menteeOnly));
  }
}

function onChangeText(props, value) {
  if (props.isMentor) {
    props.dispatch(updateMentorSummary(value));
  } else {
    props.dispatch(updateMenteeSummary(value));
  }
}

function Summary(props) {
  return (
    <View style={styles.container}>
      <View>
        <Text style={styles.headerText}>{props.headerText}</Text>
        <TextInput
          style={styles.inputStyle}
          onChangeText={value => onChangeText(props, value)}
          maxLength={500}
          multiline
          onSubmitEditing={Keyboard.dismiss}
          onEndEditing={Keyboard.dismiss}
          testID="sum-input"
        />
      </View>
      <View>
        {props.isMentor ? (
          <Button
            style={
              props.mentorSummary !== ''
                ? styles.buttonStyle
                : { ...styles.buttonStyle, backgroundColor: '#e9e9e9' }
            }
            onPress={() =>
              submit(
                props.dispatch,
                props.isMentor,
                props.mentorSummary,
                props.loadMenteeScreenNext
              )}
            disabled={props.mentorSummary === ''}
            testID="sum-submit-button">
            <Text style={styles.textStyle}>Continue</Text>
          </Button>
        ) : (
          <Button
            style={
              props.menteeSummary !== ''
                ? styles.buttonStyle
                : { ...styles.buttonStyle, backgroundColor: '#e9e9e9' }
            }
            onPress={() =>
              submit(
                props.dispatch,
                props.isMentor,
                props.menteeSummary,
                null,
                props.menteeOnly
              )}
            disabled={props.menteeSummary === ''}
            testID="sum-submit-button">
            <Text style={styles.textStyle}>Continue</Text>
          </Button>
        )}
      </View>
    </View>
  );
}

export default Summary;
