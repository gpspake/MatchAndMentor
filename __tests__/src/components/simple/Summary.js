/* eslint-env jest */
'use strict'
import React from 'react'
import Summary from '../../../../src/components/simple/Summary'
import renderer from 'react-test-renderer'
import { findById } from '../../../../setupTests'

jest.mock('TouchableOpacity', () => {
  const mockComponent = require('jest-react-native')
  return mockComponent('TouchableHighlight')
})

it('renders correctly and is a mentor', () => {
  const tree = renderer.create(<Summary dispatch={jest.fn()} headerText='mentor summary' isMentor loadMenteeScreenNext />)
  expect(tree).toMatchSnapshot()
})

it('renders correctly and is a mentee', () => {
  const tree = renderer.create(<Summary dispatch={jest.fn()} headerText='mentee summary' isMentor={false} />)
  expect(tree).toMatchSnapshot()
})

it('calls submit on button press with mentor', () => {
  const tree = renderer.create(<Summary dispatch={jest.fn()} headerText='mentor summary' isMentor loadMenteeScreenNext />)
  const submitButton = findById(tree.toJSON(), 'sum-submit-button')
  submitButton.props.onPress()
  expect(tree.toTree().props.dispatch).toBeCalled()
})

it('calls submit on button press with mentee', () => {
  const tree = renderer.create(<Summary dispatch={jest.fn()} headerText='mentor summary' isMentor={false} />)
  const submitButton = findById(tree.toJSON(), 'sum-submit-button')
  submitButton.props.onPress()
  expect(tree.toTree().props.dispatch).toBeCalled()
})

it('calls onChangeText with mentor', () => {
  const tree = renderer.create(<Summary dispatch={jest.fn()} headerText='mentor summary' isMentor loadMenteeScreenNext />)
  const input = findById(tree.toJSON(), 'sum-input')
  input.props.onChangeText()
  expect(tree.toTree().props.dispatch).toBeCalled()
})

it('calls onChangeText with mentee', () => {
  const tree = renderer.create(<Summary dispatch={jest.fn()} headerText='mentor summary' isMentor={false} />)
  const input = findById(tree.toJSON(), 'sum-input')
  input.props.onChangeText()
  expect(tree.toTree().props.dispatch).toBeCalled()
})
