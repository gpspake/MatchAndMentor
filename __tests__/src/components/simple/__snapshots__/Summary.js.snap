// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly and is a mentee 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "space-between",
      "padding": 20,
    }
  }
>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "black",
          "fontFamily": "Montserrat-Light",
          "fontSize": 24,
        }
      }
    >
      mentee summary
    </Text>
    <TextInput
      maxLength={500}
      multiline={true}
      onChangeText={[Function]}
      onEndEditing={[Function]}
      onSubmitEditing={[Function]}
      style={
        Object {
          "backgroundColor": "#e9e9e9",
          "borderBottomColor": "black",
          "borderBottomWidth": 1,
          "fontFamily": "Montserrat-Light",
          "fontSize": 24,
          "height": "50%",
          "margin": 20,
          "padding": 10,
        }
      }
      testID="sum-input"
    />
  </View>
  <View>
    <TouchableHighlight
      activeOpacity={0.85}
      disabled={false}
      onPress={[Function]}
      style={
        Object {
          "backgroundColor": "#55A838",
          "borderRadius": 10,
          "height": 45,
          "justifyContent": "center",
          "margin": 20,
        }
      }
      testID="sum-submit-button"
      underlayColor="black"
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "alignSelf": "center",
            "color": "white",
            "fontFamily": "Montserrat-Regular",
            "fontSize": 24,
          }
        }
      >
        Continue
      </Text>
    </TouchableHighlight>
  </View>
</View>
`;

exports[`renders correctly and is a mentor 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "space-between",
      "padding": 20,
    }
  }
>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "black",
          "fontFamily": "Montserrat-Light",
          "fontSize": 24,
        }
      }
    >
      mentor summary
    </Text>
    <TextInput
      maxLength={500}
      multiline={true}
      onChangeText={[Function]}
      onEndEditing={[Function]}
      onSubmitEditing={[Function]}
      style={
        Object {
          "backgroundColor": "#e9e9e9",
          "borderBottomColor": "black",
          "borderBottomWidth": 1,
          "fontFamily": "Montserrat-Light",
          "fontSize": 24,
          "height": "50%",
          "margin": 20,
          "padding": 10,
        }
      }
      testID="sum-input"
    />
  </View>
  <View>
    <TouchableHighlight
      activeOpacity={0.85}
      disabled={false}
      onPress={[Function]}
      style={
        Object {
          "backgroundColor": "#55A838",
          "borderRadius": 10,
          "height": 45,
          "justifyContent": "center",
          "margin": 20,
        }
      }
      testID="sum-submit-button"
      underlayColor="black"
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "alignSelf": "center",
            "color": "white",
            "fontFamily": "Montserrat-Regular",
            "fontSize": 24,
          }
        }
      >
        Continue
      </Text>
    </TouchableHighlight>
  </View>
</View>
`;
